tests/http_proxies/run.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -eu
+
+cur=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+source $cur/../_utils/test_prepare
+
+WORK_DIR=$TEST_DIR/$TEST_NAME
+
+function test_dm() {
+	env_name=$1
+	env_val=$2
+	export $env_name=$env_val
+
+	expected_str="\[\"using proxy config\"\] \[$env_name=$env_val\]"
+
+	# run dm master
+	run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml
+	check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT
+
+	# check dm master log
+	check_log_contains "$WORK_DIR/master/log/dm-master.log" $expected_str 1
+
+	# rum dm worker
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+
+	# check dm worker log
+	check_log_contains "$WORK_DIR/worker1/log/dm-worker.log" $expected_str 1
+
+	unset $env_name
+}
+
+function run() {
+	echo "test dm grpc proxy env setting checking for http_proxy=http://127.0.0.1:8080"
+	test_dm "http_proxy" "http://127.0.0.1:8080"
```

Comment:
The issue seems to mean that grpc will use the system default environment variables.

So, maybe you can obtain them and input them here?

Reply:
he set environment variables in this test 🤔

Reply:
Sorry, I didn't see it just now....